review/fileToReview.js

```diff
@@ -1,73 +1,119 @@
 
-var domain = "bank.local.fr"
+const winston = require('winston')
+
+const logger = winston.createLogger({
+  defaultMeta: { service: 'bankin-test' },
+  transports: [
+    new winston.transports.Console({
+      format: winston.format.json(),
+    })
+  ],
+})
+
+const domain = 'bank.local.fr'
+
+/**
+ * @description Check if response is success
+ * @param {Object} response http response
+ */
+const isSuccessResponse = (response) => response && code === 200 && response.data
+
+/**
+ * @description Check paginated request reach end
+ * @param {Object} response http response
+ */
+const hasNextPage = (response) => response.data.meta && response.data.meta.hasPageSuivante
+
+/**
+ * @description Check if data is correctly typed
+ * @param {unknow} mouvements Check if mouvements is valid list of transaction
+ */
+const assertTransactions = (mouvements) => !Array.isArray(mouvements)
+
+/**
+ * CustomError extends from Error
+ * @statusCode {string}
+ * @rawError {object} Initial error
+ * @function {string} Function who raised the error
+ */
+class CustomError extends Error {
+  constructor(info) {
+    super(...info)
+    this.name = 'Custom error'
+  }
+}
 
 /**
  * @description Fetch transactions recursively
  * @param {string} fromDate The maximum date of transactions to return
- * @param {string} authorization Authorization header to authent the user
+ * @param {string} authorization Authorization header to authenticate the user
  * @param {jws} jws Jws token, mandatory if we get one from login request
  * @param {Number} id Account id
- * @param {Number} page Number of the page
- * @param {Object} previousTransactions Previous page of transactions (To ckeck for dupes)
+ * @param {Number} page Current page number
+ * @param {Object} previousTransactions Previous page of transactions (To check for duplicates)
  * @return {Object} All transactions available on the page
  */
-async function fetchTransactions(fromDate, authorization, jws = null, id, page, previousTransactions) {
-	console.log(`--- Fetch Trasactions page n°${page} ---`);
+const fetchTransactions = async (
+  fromDate,
+  authorization,
+  jws = null,
+  id,
+  page,
+) => {
```

Comment:
Too many argument, i would prefer to put only one per line. It should be better in object.